### PR TITLE
[RISCV] Add missing RISCVMaskedPseudo for TIED pseudos

### PR DESCRIPTION
**Pull request by lukel97**
This was preventing us from folding away the vmerge into its mask.

### CODE REVIEW

**Review by wangpc-pp — APPROVED** [review]
LGTM.
I don't know if we need it for `VPseudoTiedBinaryRoundingMode`.

File: llvm/test/CodeGen/RISCV/rvv/rvv-peephole-vmerge-vops.ll [hunk 1]
```diff
@@ -1192,11 +1192,8 @@ define <vscale x 2 x i32> @vmerge_larger_vl_false_becomes_tail(<vscale x 2 x i32
 define <vscale x 2 x i64> @vpmerge_vwsub.w_tied(<vscale x 2 x i64> %passthru, <vscale x 2 x i64> %x, <vscale x 2 x i32> %y, <vscale x 2 x i1> %mask, i32 zeroext %vl) {
```
Comment (sun-jacobi) [inline llvm/test/CodeGen/RISCV/rvv/rvv-peephole-vmerge-vops.ll:1192]: Might not related to this patch but is the `%x` necessary here?

**Review by sun-jacobi — APPROVED** [review]

File: llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td [hunk 2]
```diff
@@ -2212,7 +2212,8 @@ multiclass VPseudoTiedBinary<VReg RetClass,
     def "_" # MInfo.MX # "_TIED": VPseudoTiedBinaryNoMask<RetClass, Op2Class,
                                                           Constraint, TargetConstraintType>;
     def "_" # MInfo.MX # "_MASK_TIED" : VPseudoTiedBinaryMask<RetClass, Op2Class,
-                                                         Constraint, TargetConstraintType>;
+                                                         Constraint, TargetConstraintType>,
+                                        RISCVMaskedPseudo<MaskIdx=2>;
```
Comment (preames) [inline llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td:2216]: You should probably add to VPseudoTiedBinaryRoundingMode as well.

**Review by preames — APPROVED** [review]
LGTM w/comment.

Note that this is a slightly larger change than it first appears.  We're not adding the annotation for *another* instance of a TIED pseudo, this is the *first* instance of the TIED pseudo.  

Looking at the existing code in performCombineVMergeAndVOps, I don't see a problem, but please make sure you audit this code closely - if you haven't already.

File: llvm/test/CodeGen/RISCV/rvv/rvv-peephole-vmerge-vops.ll (hunk 1 above)
Comment (lukel97) [reply llvm/test/CodeGen/RISCV/rvv/rvv-peephole-vmerge-vops.ll:1192]: Woops, it's not. Will remove it after this PR!

**Pull request merged**